pkg/cluster/storage.go

```diff
@@ -180,3 +181,41 @@ func normalizeToken(token string) (string, error) {
 	}
 	return password, nil
 }
+
+// migrateEmptyStringToken will list all keys that has prefix /bootstrap and will check for key that is
+// hashed with empty string and then migrate this empty string key and resave it with the right token
+func (c *Cluster) migrateEmptyStringToken(ctx context.Context, bootstrapList []client.Value, storageClient client.Client, emptyStringKey, tokenKey, token string) error {
+	for _, bootstrapKV := range bootstrapList {
+		// checking for empty string bootstrap key
+		if string(bootstrapKV.Key) == emptyStringKey {
+			logrus.Warn("bootstrap data encrypted with empty string, deleting and resaving with token")
+			// making sure that the process is autonmous by decrypting and rencrypting the data first
+			data, err := decrypt("", bootstrapKV.Data)
+			if err != nil {
+				return err
+			}
+			encryptedData, err := encrypt(token, data)
+			if err != nil {
+				return err
+			}
+			// saving the new encrypted data with the right token key
+			if err := storageClient.Create(ctx, tokenKey, encryptedData); err != nil {
+				if err.Error() == "key exists" {
+					logrus.Warnln("bootstrap key exists; please follow documentation on updating a node after snapshot restore")
```

Comment:
If the tokenKey already exists in our bootstrapList we should not try to create one, but in case another server has already created one we should still try to remove the emptyStringKey (and probably warn on error).